src/main/jbake/content/docs/components/metrics/index.adoc

```diff
@@ -1258,7 +1258,380 @@ A reset event occurs some time between 90000 and 210000; consequently, we will g
+range into four buckets. A higher value increases the number of buckets which in turn reduces the number of data points
+per bucket.
+
+`bucketDuration` is a duration in milliseconds that specifies how wide or large each bucket is. For the preceding
+example, `bucketDuration=900000` specifies that each bucket is 15 minutes wide, which again yields four buckets. A
+larger duration results in fewer buckets with more data points per bucket. A smaller duration results in more buckets
+with less data points per bucket.
+
+==== Numeric Bucket Data Points
+Numeric bucket data points are used with gauges, counters, and rates. When data points are grouped into a bucket,
+several aggregation functions are applied to produce a data point that consists of a number of statistics.
+
+[source,javascript]
+.Numeric bucket data point
+----
+{
+  "start": 12345,
+  "end": 6789,
+  "empty": false,
+  "min": 100.01,
+  "avg": 107.5,
+  "max": 115.32,
+  "median": 109.0,
+  "sum": 215.0,
+  "samples": 5
+}
+----
+
+`start` and `end` correspond to the bucket's start and end times respectively.
+
+`empty` is a boolean flag that indicates whether or not the bucket has any data points in it. We will see an example of
+an empty bucket next.
+
+The `min`, `max`, `avg`, `median`, and `sum` properties should be self-explanatory. They hold the results of the
+aggregation functions applied over all the data points in the bucket.
+
+`samples` is the total number of data points in the bucket.
+
+The properties in a numeric data point are fixed and are the same for gauges, counters, and rates.
+
+NOTE: In the future, Hawkular Metrics may allow the client to specify which aggregation functions to use in the bucket
+data points. See link:https://issues.jboss.org/browse/HWKMETRICS-374[HWMKETRICS-374] for details.
+
+Now let's see what an empty bucket data point looks like.
+
+[source,javascript]
+.Empty numeric bucket data point
+----
+{
+  "start": 12345,
+  "end": 6789,
+  "empty": true,
+  "min": "NaN",
+  "avg": "NaN",
+  "max": "NaN",
+  "median": "NaN",
+  "sum": "NaN",
+  "samples": 0
+}
+----
+
+The `empty` property is true indicating that there were no data points in the bucket. The `NaN` values correspond to
+Java's link:https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#NaN[Double.NaN].
+
+A bucket data point can also have an optional set of percentiles.
+
+[source,javascript]
+.Bucket data point with percentiles
+----
+{
+  "start": 12345,
+  "end": 6789,
+  "empty": false,
+  "min": 100.01,
+  "avg": 107.5,
+  "max": 115.32,
+  "median": 109.0,
+  "sum": 215.0,
+  "percentiles": [
+    {
+      "quantile": 0.90,
+      "value": 100.01
+    },
+    {
+      "quantile": 0.95
+      "value": 108.42
+    },
+    {
+      "quantile": 0.99
+      "value": 115.25
+    }
+  ]
+  "samples": 5
+}
+----
+
+This data point includes the 90th, 95th, and 99th percentiles. Unless the request explicitly asks for percentiles, they
+will be omitted.
```

Comment:
I suppose so. Maybe just a sentence like "more details below"?

2016-04-28 16:05 GMT+02:00 jsanda <notifications@github.com>:

> In src/main/jbake/content/docs/components/metrics/index.adoc
> <https://github.com/hawkular/hawkular.github.io/pull/165#discussion_r61432267>
> :
>
> > +    },
> > +    {
> > +      "quantile": 0.95
> > +      "value": 108.42
> > +    },
> > +    {
> > +      "quantile": 0.99
> > +      "value": 115.25
> > +    }
> > +  ]
> > +  "samples": 5
> > +}
> > +----
> > +
> > +This data point includes the 90th, 95th, and 99th percentiles. Unless the request explicitly asks for percentiles, they
> > +will be omitted.
>
> The following section has an example. Is that sufficient?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/hawkular/hawkular.github.io/pull/165/files/5a234b1bb0d9587b2fe4b696c6621f6f15fabce2#r61432267>
>



-- 
Thomas Segismont
JBoss ON Engineering Team